source/postprocess/visualization/darcy_velocity.cc

```diff
@@ -0,0 +1,135 @@
+/*
+  Copyright (C) 2011 - 2024 by the authors of the ASPECT code.
+
+  This file is part of ASPECT.
+
+  ASPECT is free software; you can redistribute it and/or modify
+  it under the terms of the GNU General Public License as published by
+  the Free Software Foundation; either version 2, or (at your option)
+  any later version.
+
+  ASPECT is distributed in the hope that it will be useful,
+  but WITHOUT ANY WARRANTY; without even the implied warranty of
+  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+  GNU General Public License for more details.
+
+  You should have received a copy of the GNU General Public License
+  along with ASPECT; see the file LICENSE.  If not see
+  <http://www.gnu.org/licenses/>.
+*/
+
+
+#include <aspect/postprocess/visualization/darcy_velocity.h>
+#include <aspect/utilities.h>
+#include <aspect/melt.h>
+#include <aspect/simulator.h>
+
+namespace aspect
+{
+  namespace Postprocess
+  {
+    namespace VisualizationPostprocessors
+    {
+      template <int dim>
+      DarcyVelocity<dim>::
+      DarcyVelocity ()
+        :
+        DataPostprocessorVector<dim> ("darcy_velocity",
+                                      update_values | update_quadrature_points),
```

Comment:
The material model also needs the gradients I think (it doesnt matter much for most models, because at least one of the postprocessors will likely need the gradients anyway, but just in case).